src/client.js

```diff
@@ -104,17 +105,34 @@ class Client {
   /**
    * Resolve the DNS shortname `name`. If `name` is already a cabal key,  it will
    * be returned and the DNS lookup is aborted.
+   * If `name` is a whisper:// key, a DHT lookup for the passed-in key will occur. 
+   * Once a match is found, it is assumed to be a cabal key, which is returned.
    * Returns the cabal key in `cb`. If `cb` is null a Promise is returned.
-   * @param {string} name the DNS shortname
-   * @param {function(string)} [cb] The callback to be called when DNS lookup succeeds
+   * @param {string} name the DNS shortname, or whisper:// shortname
+   * @param {function(string)} [cb] The callback to be called when lookup succeeds
    */
   resolveName (name, cb) {
-    return this.cabalDns.resolveName(name).then((key) => {
-      if (key === null) return null
-      key = Client.scrubKey(key)
-      if (!cb) return key
-      else cb(key)
-    })
+    if (name.startsWith("whisper://")) {
+        return new Promise((resolve, reject) => {
+            let slip = ""
+            const stream = paperslip.read(name.slice(10))
+            stream.on("data", (datum) => {
```

Comment:
`stream.on('end')` isn't the only way a stream can end unfortunately; you could use a combination of `concat-stream` and `pump` to do this more reliably.